cmd/libs/go2idl/make-client/generators/client-generator.go

```diff
@@ -0,0 +1,270 @@
+		PackagePath: arguments.OutputPackagePath,
+		HeaderText: append(boilerplate, []byte(
+			`
+// This file was autogenerated by the command:
+// $ `+strings.Join(os.Args, " ")+`
+// Do not edit it manually!
+
+`)...),
+		PackageDocumentation: []byte(
+			`// Package unversionedClient has the automatically generated clients for unversioned resources.
+`),
+		// GeneratorFunc returns a list of generators. Each generator makes a
+		// single file.
+		GeneratorFunc: func(c *generator.Context) (generators []generator.Generator) {
+			generators = []generator.Generator{
+				// Always generate a "doc.go" file.
+				generator.DefaultGen{OptionalName: "doc"},
+			}
+			// Since we want a file per type that we generate a set for, we
+			// have to provide a function for this.
+			for _, t := range c.Order {
+				generators = append(generators, &genClient{
+					DefaultGen: generator.DefaultGen{
+						// Use the privatized version of the
+						// type name as the file name.
+						//
+						// TODO: make a namer that converts
+						// camelCase to '-' separation for file
+						// names?
+						OptionalName: c.Namers["private"].Name(t),
+					},
+					outputPackage: arguments.OutputPackagePath,
+					typeToMatch:   t,
+					imports:       generator.NewImportTracker(),
+				})
+			}
+			return generators
+		},
+		FilterFunc: func(c *generator.Context, t *types.Type) bool {
+			_, found := topLevelTypes[t]
+			return found
+		},
+	}}
+}
+
+// genClient produces a file with a set for a single type.
+type genClient struct {
+	generator.DefaultGen
+	outputPackage string
+	typeToMatch   *types.Type
+	imports       *generator.ImportTracker
+}
+
+// Filter ignores all but one type because we're making a single file per type.
+func (g *genClient) Filter(c *generator.Context, t *types.Type) bool { return t == g.typeToMatch }
+
+func (g *genClient) Namers(c *generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer(g.outputPackage, g.imports),
+	}
+}
+
+func (g *genClient) Imports(c *generator.Context) (imports []string) {
+	return append(g.imports.ImportLines())
+}
+
+// GenerateType makes the body of a file implementing a set for type t.
+func (g *genClient) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
+	sw := generator.NewSnippetWriter(w, c, "$", "$")
+	pkg := filepath.Base(t.Name.Package)
+	m := map[interface{}]interface{}{
+		"type":    t,
+		"package": pkg,
+		"Package": namer.IC(pkg),
+		// @lavalamp: how about making a plural function in pkg/util, RESTMapper also needs such a function.
+		// m["plural"] = util.Plural(type.Name.Name)
+		"fieldSelector":    fieldSelector,
+		"labelSelector":    labelSelector,
+		"watchInterface":   watchInterface,
+		"apiDeleteOptions": apiDeleteOptions,
+		"apiListOptions":   apiListOptions,
+	}
+	sw.Do(nameSpacerTemplate, m)
+	sw.Do(interfaceTemplate, m)
+	sw.Do(structTemplate, m)
+	sw.Do(newStructTemplate, m)
+	sw.Do(listTemplate, m)
```

Comment:
nit: to be honest, I would change the order of the methods to:
Create
Update
Delete
Get
List
Watch
// possibly others?

I know why you have that order (it's the same as in Deployment), but we have a different order in every file, so we need to unify it anyway, and if we change that anyway, having something more intuitive would be nice.
WDYT?